dotcom-rendering/src/model/extractTrendingTopics.ts

```diff
@@ -2,79 +2,96 @@ import { isNonNullable } from '@guardian/libs';
 import type { FECollectionType, FEFrontCard } from '../types/front';
 import type { FETagType } from '../types/tag';
 
+/**
+ * Extracts sectionId from a given tag if its a section tag
+ * @param tag
+ * @returns a section ID or undefined
+ */
+const extractTagSectionId = (tag: FETagType): string | undefined => {
+	if (tag.properties.id === 'uk/uk') {
+		return 'uk-news';
+	}
+
+	const split = tag.properties.id.split('/');
+
+	return split[0] === split[1] ? split[0] : undefined;
+};
+
+/**
+ * Checks if a given tag is a section ID tag
+ * @param tag
+ * @returns a section ID or undefined
+ */
+const isNotSectionTag = (tag: FETagType): boolean => {
+	const sectionId = extractTagSectionId(tag);
+
+	return !(
+		sectionId !== undefined &&
+		tag.properties.sectionId !== undefined &&
+		sectionId.includes(tag.properties.sectionId)
+	);
+};
+
 /**
  * Gets all relevant tags filtered by properties
  * @param tags - The deduplicated trails
  * @returns An array of relevant tags
  */
-const getTags = (trails: FEFrontCard[]): FETagType[] =>
+const getTags = (trails: FEFrontCard[], pageId: string): FETagType[] =>
 	trails
 		.flatMap((trail) => trail.properties.maybeContent?.tags.tags)
 		.filter(isNonNullable)
+		.filter(isNotSectionTag)
+		.filter((tag) => tag.properties.id !== pageId)
 		.filter((tag) => {
 			return (
-				tag.properties.paidContentType?.includes('Keyword') ??
-				tag.properties.paidContentType?.includes('Topic') ??
+				!!tag.properties.paidContentType?.includes('Keyword') ||
+				!!tag.properties.paidContentType?.includes('Topic') ||
 				tag.properties.tagType === 'Keyword'
 			);
 		});
 
 /**
- * Gets tags which have the 5 most used IDs from the provided tags
+ * Sort tags by frequency
  * @param tags - Tags which you want to have filtered
- * @returns An array of tags which has been filtered to only include those with the most popular tag IDs
+ * @returns An array of tags which has been sorted by frequency
  */
-const filterTopFive = (tag: FETagType[]): FETagType[] => {
-	const idCounts: { [key: string]: number } = {};
+const sortTags = (tag: FETagType[]): FETagType[] => {
+	const counter: { [key: string]: [FETagType, number] } = {};
 
 	tag.forEach((x) => {
 		const id = x.properties.id;
-		if (idCounts[id] !== undefined) {
-			idCounts[id]++;
-		} else {
-			idCounts[id] = 1;
-		}
+		const count = counter[id]?.[1] ?? 0;
+
+		counter[id] = [x, count + 1];
 	});
 
-	const topFiveIds = Object.entries(idCounts)
-		.sort((a, b) => b[1] - a[1])
-		.slice(0, 5)
-		.map((entry) => entry[0]);
+	const sortedTags = Object.entries(counter).sort(
+		(a, b) => b[1][1] - a[1][1],
+	);
 
-	return tag.filter((x) => topFiveIds.includes(x.properties.id));
+	return sortedTags.map((x) => x[1][0]);
 };
```

Comment:
This may be clearer?

```ts
/**
 * Sort tags by frequency
 * @param tags - Tags which you want to have filtered
 * @returns An array of tags which has been sorted by frequency
 */
const sortTags = (tags: readonly FETagType[]): FETagType[] => {
	const map = new Map<string, number>();

	for (const tag of tags) {
		const {
			properties: { id },
		} = tag;
		const count = map.get(id) ?? 0;
		map.set(id, count + 1);
	}

	const getCount = (tag: FETagType) => map.get(tag.properties.id) ?? -1;

	return tags.slice().sort((a, b) => getCount(b) - getCount(a));
};
```

Reply:
```
return tags.slice().sort((a, b) => getCount(b) - getCount(a));
```

I think this wouldn't de-dupe the tags

Reply:
Ah yes, you would want a `Map` for that:

```ts
/**
 * Sort tags by frequency
 * @param tags - Tags which you want to have filtered
 * @returns An array of tags which has been sorted by frequency
 */
/**
 * Sort tags by frequency
 * @param tags - Tags which you want to have filtered
 * @returns An array of tags which has been sorted by frequency
 */
const sortTags = (tags: readonly FETagType[]): FETagType[] => {
	const map = new Map<string, { count: number; tag: FETagType }>();

	for (const tag of tags) {
		const {
			properties: { id },
		} = tag;

		map.set(id, { count: (map.get(id)?.count ?? 0) + 1, tag });
	}

	return [...map.values()]
		.sort(({ count: a }, { count: b }) => b - a)
		.map(({ tag }) => tag);
};
```